src/index.js

```diff
@@ -57,14 +65,20 @@ const getTextNodesInElementContainingText = (element, text) => {
 const getRangesForSearchTermInElement = (element, search) => {
   const ranges = [];
   const lowerCaseSearch = search.toLowerCase();
-  if (!element.firstChild) return ranges;
-  const text = element.textContent?.toLowerCase() || "";
+  if (!element.childNodes.length === 0) return ranges;
+  // In some frameworks like React, when combining static text with dynamic text, the element may have multiple Text child nodes.
+  // To avoid errors, we must find the child node that actually contains the search term.
+  const childWithSearchTerm = Array.from(element.childNodes).find((node) =>
+    node.textContent?.toLowerCase().includes(lowerCaseSearch)
+  );
+  if (!childWithSearchTerm) return ranges;
```

Comment:
do you still need the try/catch in the main function if you have this safety here?

Reply:
I believe we should keep it anyway